### DISCUSSION TITLE
Un archivio di joypad?

### DISCUSSION THREAD

**Discussion by bionicco** [pdonadeo/e-uscito-joypad#5] [Ideas]
Ciao e grazie del lavoro svolto.
Da qualche giorno volevo chiedere il tuo contatto per fare una proposta su un'idea che mi gira in testa da un po'.
Mi piacerebbe fare una archivio di Joypad, in modo da poter andare a ritrovare gli episodi in cui si è parlato di un determinato titolo (perchè cercare una recensione è troppo facile, è più bello ascoltare un episodio del podcast)
Ovviamente mettendomi a disposizione per sviluppare la cosa (o produrre un po' di dati) anche se non avrei voglia/tempo di studiare OCaml o molto altro che non sia angular in questo periodo.

**Comment by pdonadeo** [comment]
@bionicco ho poco tempo ma l'idea mi piace: ti rispondo con un po' di precisione tra qualche giorno.

**Comment by bionicco** [comment]
Benissimo! Anche il mio tempo è piuttosto limitato, ma non c'è nessuna fretta.
Aspetto aggiornamenti e se preferisci contattami pure via email

**Comment by pdonadeo** [comment]
@bionicco per discutere preferisco in pubblico invece che tramite email privata, così la storia delle scemenze che spariamo rimane per tutti.

Per questo con spostato la _issue_ che avevi aperto dentro le _Discussions_ di GitHub.

## L'idea

Nel merito: l'idea dell'archivio mi piace ma va concretizzata un attimo, se no giriamo a vuoto. Quello che hai scritto mi fa venire in mente almeno due cose:

1. un semplice elenco di tutti i giochi mai citati in Joypad associati all'episodio e magari al minutaggio preciso insieme ad un semplice campo di ricerca in cui scrivendo il nome del gioco appaiono gli episodi ed il minuto in cui è stato citato.
2. una vera e propria ricerca full text in tutti i testi di Joypad.

Il primo è realizzabile completamente lato client in Javascript, al massimo con l'ausilio di una API lato server se non vogliamo infilare tutti i dati in un file statico. Ma stiamo parlando di pochi dati, si può fare in entrambi i modi.

Il secondo è più complicato: serve un database con capacità di ricerca _full text_ come PostgreSQL o Elastic. Non che non si possa fare, ci mancherebbe, diciamo che inizia ad avere un costo.

## Requisiti

In entrambi i casi c'è del lavoro da fare, e parecchio.

Nel primo caso occorre che qualcuno (tu, io o altri disperati che riuscissimo a coinvolgere nel progetto) si riascolti **tutti** gli episodi e trascriva i nomi dei giochi, col minutaggio. A quel punto non è difficile prendere questa roba, metterla in un file di qualche tipo e servirla al frontend, che ha un piccolo campo di ricerca.

Nel secondo caso... a parte mettere in piedi un DB full text occorre avere il transcript di tutti gli episodi del podcast. A meno di non farlo con sistemi automatici (ma non saprei) penso sia impossibile: chi si "sbobina" a mano 47 ore di parlato?

## Limiti e problemi

In ogni caso vedo qualche problema legale nel fare tutto questo: serve avere una approvazione  scritta dall'editore (Il Post) per usare logo, testi, eccetera. Credo che non sia un vero problema e credo che possa solo far piacere ma non è detto: magari hanno in mente servizi analoghi per tutti i loro podcast ed ovviamente vorranno ospitare questi servizi sul loro sito. Oppure solo per gli abbonati. Meglio chiarire questi punti con loro.

Intanto nulla ci vieta di procedere a ragionare so cosa vogliamo o possiamo fare.

**Reply by bionicco** [reply]
La mia idea era semplicemente la prima, anche perchè la seconda mi sembra più difficilmente realizzabile sotto vari aspetti.
Quindi quello che pensavo era un'interfaccia che permetta di:
- fare una ricerca per titolo e vedere in quali episodi era stato nominato, magari anche quando, da chi e in che forma (consiglio, recensione,etc)
- avere dualmente una lista degli episodi con i titoli (e volendo argomenti) trattati
Il tutto potrebbe stare in un paio di json e come dicevi basta capire dove metterli. 

Per realizzare la cosa i passi potrebbero essere i seguenti:
1. Definire le informazioni che si vorrebbero pubblicare
2. Confrontarsi con Zampa (o gli altri) sulla fattibilità della cosa, cioè se possiamo essere autorizzati e se ci supportano nel cercare volontari
3. Definire un modo per dividerci la mappatura degli episodi e un sistema per raccogliere queste informazioni
4. Coinvolgere gli ascoltatori che desiderano aiutare e raccogliere i dati
5. Valutare eventuali arricchimenti dei dati (link alle puntate, ulteriori dati dei giochi, etc)

Ovviamente in parallelo una volta che è stata appurata la fattibilità (quindi dopo il punto 2) c'è la creazione del frontend (e dell'eventuale API)
PS: in allegato un possibile formato dei dati da editare/commentare/completare
[joypadarchive_example.json.txt](https://github.com/pdonadeo/e-uscito-joypad/files/8502505/joypadarchive_example.json.txt)

**Comment by pdonadeo** [comment]
Li sento io, vediamo che margini ci sono.

**Reply by agofa** [reply]
Mi presento. Sono un potenziale disperato ascoltatore e appuntatore di nome gioco e minutaggio...

**Reply by bionicco** [reply]
> Li sento io, vediamo che margini ci sono.

Qualche novità?

**Reply by pdonadeo** [reply]
Purtroppo no 😢

**Comment by pdonadeo** [comment]
Mi ha risposto Zampa: per Il Post non ci sono problemi a procedere. Dice anche che potrà aiutare, nel limite del possibile.

Gli ho chiesto se hanno una scaletta scritta delle puntate, per ricavare i primi dati.

Vi farò sapere.

**Reply by agofa** [reply]
La scaletta in genere è oggetto del contendere durante le puntate. Episodi. Insomma quella cosa lì....
Quindi qualcosa dovrebbe esistere.

**Reply by zampale** [reply]
In effetti essa esiste e posso fornirvele pure. Devo solo capire come avendo seicento anni e non volendo fare le foto dello schermo per poi mandarvele come i nonni su Whatsapp

**Reply by pdonadeo** [reply]
> In effetti essa esiste e posso fornirvele pure. Devo solo capire come avendo seicento anni e non volendo fare le foto dello schermo per poi mandarvele come i nonni su Whatsapp

@zampale non sono in un qualche file? Sono scritte a mano?

**Comment by agofa** [comment]
@pdonadeo e @zampale in effetti potrebbero farmi comodo le scalette, essendo un po' sordo (non è una battuta) faccio a volte fatica a capire i nomi di giochi e autori.
Si può avere accesso? Account "Gogol" è pindus@gmail.com

**Reply by pdonadeo** [reply]
Te le mando subito.

**Reply by agofa** [reply]
Grazie
Ho visto che ci sono le copertine delle puntate, almeno nelle due che ho aperto a caso
Potremmo usare quelle...
Anzi troppa grazia, 2 o 3 immagini per ogni punt..epis..insomma

**Reply by agofa** [reply]
Ma solo per stagione 2 e 3, non la 1
